config/v1/types_oauth.go

```diff
@@ -29,11 +29,16 @@ type OAuthSpec struct {
 	// templates allow you to customize pages like the login page.
 	// +optional
 	Templates OAuthTemplates `json:"templates"`
+
+	// masterAuthConfig contains options altering the auth configuration of the kube api server
+	// +optional
+	MasterAuthConfig *MasterAuthConfig `json:"masterAuthConfig"`
```

Comment:
`OAuthSpec` is about how we configure a managed oauth server. This type is about how to configure authentication on the cluster.  I don't think they belong together.  Why wouldn't this be in an authentication type?

Reply:
Yeah I can move both there.